sources/packages/backend/apps/api/src/route-controllers/designation-agreement/designation-agreement.institutions.controller.ts

```diff
@@ -105,9 +105,9 @@ export class DesignationAgreementInstitutionsController extends BaseController {
     const createdDesignation =
       await this.designationAgreementService.submitDesignationAgreement(
         userToken.authorizations.institutionId,
-        payload.dynamicData,
+        submissionResult.data.dynamicData,
```

Comment:
These values are being consumed as `any`. We need to type them in some way.
Maybe we can change the `DryRunSubmissionResult` as below:
```ts
export interface DryRunSubmissionResult<T = unknown> {
  valid: boolean;
  data?: T;
  formName: string;
}
```
The `filter((location)` and the `map((location)` are treating the `location` and `any` also due to this.
If not in this PR I believe we should do one more PR right after this and maybe also change the `data` property to `resultData` to avoid the `data.data`.

@guru-aot @dheepak-aot FYI

Reply:
For now I just typed the dryrun submission result. We need to talk about it. Thanks!

Reply:
I made the `DryRunSubmissionResult` optional generic. Then when needed/possible we can type the `.data.data` object.